src/main/java/org/commcare/formplayer/services/RestoreFactory.java

```diff
@@ -429,20 +424,27 @@ public boolean isRestoreXmlExpired() {
         }
     }
 
-    public InputStream getRestoreXml() {
-        return getRestoreXml(false);
-    }
-
     public InputStream getRestoreXml(boolean skipFixtures) {
         ensureValidParameters();
-        Pair<URI, HttpHeaders> restoreUrlAndHeaders = getRestoreUrlAndHeaders(skipFixtures);
-        recordSentryData(restoreUrlAndHeaders.first.toString());
-        log.info("Restoring from URL " + restoreUrlAndHeaders.first.toString());
-        InputStream restoreStream = getRestoreXmlHelper(restoreUrlAndHeaders.first, restoreUrlAndHeaders.second);
+        URI url = getRestoreUrl(skipFixtures);
+        recordSentryData(url.toString());
+        log.info("Restoring from URL " + url);
+        InputStream restoreStream = getRestoreXmlHelper(url);
         setLastSyncTime();
         return restoreStream;
     }
 
+    public HttpHeaders getRequestHeaders(URI url) {
+        HttpHeaders headers;
+        if (getHqAuth() == null) {
```

Comment:
It looks like Hq auth is expected to be null mostly for the sms requests. If I am right about that, it seems like the case restore requests in a normal (non-sms) flow will now won't have a Hmac auth unlike before when they were all Hmac. Do you see any issue with that ?

Reply:
> It looks like Hq auth is expected to be null mostly for the sms requests

As far as I know this is only for SMS workflows and never happens in normal user workflows.

> If I am right about that, it seems like the case restore requests in a normal (non-sms) flow will now won't have a Hmac auth unlike before when they were all Hmac.

Where are case restores coming from outside of SMS workflows? 

> Do you see any issue with that ?

Even if that is the case I think we should always prefer use auth over HMAC auth if it is available. I do see however that the case restores endpoint only accepts HMAC auth so if user auth is ever present then that will be an issue.

Reply:
Sounds good. I didn't realise that case restores are specific to sms workflows.